Add Tensorflow Lite Micro as a module

laurenmurphyx64 opened the pull request:
Adds TensorFlow Lite Micro as a module and makes some updates to reflect the change from using the larger TensorFlow repo to the new, smaller TensorFlow Lite Micro repo. Cannot be merged until [the module PR enabling Zephyr support](https://github.com/zephyrproject-rtos/tflite-micro/pull/1) is merged.

Fixes  #36947

laurenmurphyx64 committed:
modules: update tensorflow to tflite-micro

Updates west manifest and module folder name to point to new
TensorFlow Lite Micro repo Google recently pulled out of
the TensorFlow repo.

Signed-off-by: Lauren Murphy <lauren.murphy@intel.com>

laurenmurphyx64 committed:
docs, samples: tflite-micro: change tensorflow to tensorflow lite micro

Changes name of folder from tensorflow to tflite-micro and updates docs
to reference TensorFlow Lite Micro specifically instead of TensorFlow.

Signed-off-by: Lauren Murphy <lauren.murphy@intel.com>

laurenmurphyx64 commented:
(Note: The build will be broken until the PR for the module is merged.)

JordanYates commented on modules/tflite-micro/CMakeLists.txt:
```diff
@@ -12,77 +12,117 @@ if(CONFIG_TENSORFLOW_LITE_MICRO)
+    ${TENSORFLOW_LITE_MICRO_DIR}/tensorflow/lite/micro/examples/hello_world/constants.cc
+    ${TENSORFLOW_LITE_MICRO_DIR}/tensorflow/lite/micro/examples/hello_world/output_handler.cc
+    ${TENSORFLOW_LITE_MICRO_DIR}/tensorflow/lite/micro/examples/hello_world/model.cc
+    ${TENSORFLOW_LITE_MICRO_DIR}/tensorflow/lite/micro/system_setup.h
```
Why are all these headers part of `zephyr_library_sources`?

laurenmurphyx64 replied:
@JordanYates Removed in most recent commit. I put them in originally since they were listed in the TensorFlow-generated CMakeLists.txt next to  the source files. Sorry about that!

nashif commented:
> (Note: The build and compliance checks will be broken until the PR for the module is merged.)

That does not need to be this way, you should actually be pointing to the PR in the west yaml file and not a specific commit. This is how we can verify that a module changes works with the Zephyr tree.

laurenmurphyx64 committed:
modules: tflite-micro: fix cmakelists.txt

Adds *_common.cc files to fix undefined references during build
of updated samples. Remove unnecessary LICENSE files. Remove
unnecessary header files in zephyr_library_sources().

Signed-off-by: Lauren Murphy <lauren.murphy@intel.com>

laurenmurphyx64 committed:
samples: tflite-micro: update hello_world

Updates tflite-micro hello_world sample code.

Signed-off-by: Lauren Murphy <lauren.murphy@intel.com>

laurenmurphyx64 committed:
samples: tflite-micro: fix magic_wand

Fixes accelerometer_handler.cc to import correct header
with TfLiteStatus since c_api_internal.h file was deleted in
the new standalone tflite-micro repo.

Removes import of version.h in main_functions.cc to match
tflite-micro version.

Signed-off-by: Lauren Murphy <lauren.murphy@intel.com>

laurenmurphyx64 commented:
Recent updates:
- Fixing module name in tflite-micro repo (see [PR](https://github.com/zephyrproject-rtos/tflite-micro/pull/2)). Will change the Zephyr repo's west.yml from the PR revision to the zephyr branch revision once the PR is merged.
- Address feedback on headers included in library sources
- Update Hello World sources

JordanYates commented on modules/tflite-micro/CMakeLists.txt:
```diff
@@ -0,0 +1,128 @@
+    ${TENSORFLOW_LITE_MICRO_DIR}/tensorflow/lite/micro/kernels/transpose_conv.cc
+    ${TENSORFLOW_LITE_MICRO_DIR}/tensorflow/lite/micro/kernels/unpack.cc
+    ${TENSORFLOW_LITE_MICRO_DIR}/tensorflow/lite/micro/kernels/zeros_like.cc
+    ${TENSORFLOW_LITE_MICRO_DIR}/LICENSE
```
Same deal with the LICENSE files, pretty sure they aren't needed to compile :)

laurenmurphyx64 replied:
Oops. That's a duh moment on my part. Removed!

chen-png commented on samples/modules/tflite-micro/hello_world/src/main_functions.cc:
```diff
@@ -104,32 +96,34 @@ void loop(void)
 	 * for each supported hardware target.
 	 */
-	HandleOutput(error_reporter, x_val, y_val);
+  HandleOutput(error_reporter, x, y);
```
this indentation seems incorrect.

laurenmurphyx64 replied:
Good catch! Fixed in latest pushed commits.

chen-png reviewed (approved):
LGTM

laurenmurphyx64 commented:
**Note:** Please don't merge until https://github.com/zephyrproject-rtos/tflite-micro/pull/2 is merged.

carlescufi commented:
> **Note:** Please don't merge until [zephyrproject-rtos/tflite-micro#2](https://github.com/zephyrproject-rtos/tflite-micro/pull/2) is merged.

It is merged now, can you update to the SHA?
